.github/workflows/build.yml

```diff
@@ -47,3 +48,126 @@ jobs:
 
     - name: Test run
       run: /usr/local/bin/rtl_airband -v
+    
+    - name: Set snapshot version
+      id: sha
+      run: echo "snapshot_version=$(script/snapshot_version.sh)" >> $GITHUB_OUTPUT
+    
+    - name: Create debian package
+      if: runner.os == 'Linux'
+      run: |
+        pkg_dir="$(mktemp --directory)"
+        export PACKAGE="rtl-airband"
+        export VERSION="${{ steps.sha.outputs.snapshot_version }}"
+        export ARCH="$(dpkg --print-architecture)"
+        export MAINTAINER="charlie-foxtrot"
+        export DEPENDS="libc6 (>= 2.34), libconfig++9v5 (>= 1.5-0.4), libmp3lame0 (>= 3.100), libshout3 (>= 2.4.5), librtlsdr0 (>= 0.6.0), libsoapysdr0.8 (>= 0.8.1), libfftw3-single3 (>= 3.3.8), libpulse0 (>= 14.2)"
+        export DESCRIPTION="RTLSDR Airband - A multiband decoder for AM and NFM signals"
+        script/package.sh "${pkg_dir}"
+    
+    # TODO use brew for macOS? https://brew.sh/
+    - name: Create macOS package
+      if: runner.os == 'macOS'
+      run: |
+        tar -zcvf rtl-airband_${{ steps.sha.outputs.snapshot_version }}_${{ matrix.os }}.tar.gz build_Release_NFM/src/rtl_airband
+    
+    - name: "Upload package as artifact"
+      uses: actions/upload-artifact@v3
+      with:
+        name: "rtl-airband_${{ steps.sha.outputs.snapshot_version }}_${{ matrix.os }}"
+        path: |
+          *.deb
+          *.tar.gz
+        retention-days: 7
```

Comment:
7 days is fine for a PR, can this be forever on a publish from `main`?

Reply:
Yes, that value can be selected w.r.t. which branch/tag is building, but in this case I don't think it is needed since it is a _build_ artifact, not a _release_ artifact.

Release artifacts are created in the last step when all artifacts are available (GitHub + foreign architectures). This last step is currently creating a _pre-release_ and it will create a _release_ in case it will build on a tag (so to release will be enough to tag a commit)